### DISCUSSION TITLE
Doorbell volume controls

### DISCUSSION THREAD

**Discussion by danielsmith89** [tsightler/ring-mqtt#316] [Feature Requests]
Hi,

First of all, great add on, works so well compared to default HA Ring integration.

I wondered if in a future release it might be possible to control the volume of my Ring Doorbell?

Thanks.

**Comment by tsightler** [comment]
Ha, well, I didn't even know the doorbell had a volume control so there's that.  Of course it's possible, but I have no idea what the command is to do so and I don't have any time to research what that command would be so I think it would be unlikely for the time being unless someone else does it.

**Comment by jampez77** [comment]
Been having a play around with MITMProxy and thought I'd check out the volume controls. Hope this helps:
<img width="632" alt="Screenshot 2023-01-19 at 09 13 11" src="https://user-images.githubusercontent.com/1823647/213402245-9a238fe1-302a-4e9c-adf9-cf24359ad500.png">

**Reply by tsightler** [reply]
Interesting, do you have any idea of all doorbells have volume control?  Also, any indication what is the range (1-10 maybe)?  Looking at the Ring website it seems there are different volume controls for different devices.  I have a Doorbell Pro and it has volume control for "Doorbell Sounds", but it seems some doorbells, as well as the Floodlight Plus/Pro, have volume control for "Device Speaker".  I'm guessing those are different API endpoints and I'd need to figure out some way to detect which devices have it and which don't in some standard way.

**Reply by jampez77** [reply]
So the range is actually 0 - 11

I'm not sure if all doorbells do support it as I can control the volume on my own doorbell but not on my mother-in-laws doorbell that we have shared access to. Of course this could be a limitation of the shared access but I won't know until I'm next there.

Interestingly both devices report `doorbell_volume` in the `settings` object so i'm not seeing an obvious way of differentiating the feature being enabled.

**Comment by Ltek** [comment]
Would love to see Volume and Mute available... the HA native integration has it. Kinda sucks to need to install the native integration for just these 2 features.

<img width="1682" height="417" alt="image" src="https://github.com/user-attachments/assets/3649fbe8-bc02-4179-909d-26ca0c0bf282" />

**Reply by tsightler** [reply]
Why does it suck as ring-mqtt was never even intended to repalce the native integration, heck, early versions did not even have camera support at all and, more likely, now that camera live streams are supported in the native integration, I see little reason to contiue to support camers in this project going forward.

**Reply by Ltek** [reply]
Ring-MQTT is much better than the native Ring integration, for cameras.  

1. Your integration exposes far more camera options & features (example: RTSP stream is VERY valuable)  and only missing Volume and Mute.
2. Your integration's Live Stream works for doorbells, and has for a long time... the native integration had live stream broken for 2 or more years (IF it ever worked) and it did not get fixed by a maintainer. A new person fixed it and we were lucky they saw the need since I tried and failed to fix the code (I'm not a good coder).. I have zero faith the native integration will get proper maintenance.

**Reply by tsightler** [reply]
The thing is, I'm not convinced this project will get proper maintenance either.  The upstream library has many updates needed due to underlying API changes and I'm temporarily using a hacked up fork of it that works mostly OK-ish, but I'm still waiting on my PRs to be merged, not that I've had time to really push for it.

And I mostly don't have Ring cameras anymore, and hopefully won't have any at all within the next 1-2 years, so I have little incentive to maintain camera support in the long term.  I haven't really found a suitable replacement for Ring Alarm, so ring-mqtt continues as-is for now, but, it's in absolute minimal maintenance mode at this point, as I just have very little time to work on it and, realistically, someone could likely code a replacement for it within an hour using an AI agent so I'm just not sure it's really needed these days.

**Reply by Ltek** [reply]
I get what you are saying... for me it works great and if it keeps working I'm happy. I have tried (multiple months) ~6 other non-Ring cameras and Ring just works far, far better in every way ... so while I'm not a fan of subscriptions, it is one of the few I have. I'm testing a 4k Ring floodlight cam starting this week (in place of my 2k Tapo long-time floodlight cam). I get a lot of false positives with the Tapo, and even with it connected to Blue Iris with AI... I'm very curious if the Ring is better. The doorbell (which sees a ton of people, animals, cars, shadows, trees, etc) is flawless with motion detection.

**Reply by tsightler** [reply]
I will consider adding these features if I ever get time to work on this project enough to implement them, I agree they are nice and should be quite simple, but no promises on timing, the biggest issue is know what devices support these features as Ring has way too many devices now.

My biggest concern is that, at some point, Ring will make changes that will break this stuff, and, in the past, I've spent my time figuring it out, but they are making this more and more difficult and I've lost interest in wanting to reverse engineer all this stuff.  Ring has been around for ages, they clearly don't care about supporting/integrating with OSS automation tools, so why should I support them if that's what I want to use.

When I made this tool, I already own Ring cameras, so the motivation was different, as I came to HA later and just wanted to make the devices I already used work with it, but now over 8 years later, it's time to replace all of that and I'm not going to choose cameras that don't care about my primary automation platform.  That being said, I'm not replacing everything I have at once and I have at least a couple of current gen Ring cameras, so I will still try to keep this working in the interim as long as it doesn't completely break.
